qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -322,6 +324,16 @@ def _contains_delay(property_set):
     else:
         _alignments = []
 
+    # Build nested Flow controllers
+
+    # Check if the gates are in the basis to determine whether to run _unroll passes
+    _unroll_check = [GatesInBasis(basis_gates)]
+
+    def _unroll_condition(property_set):
+        return not property_set["all_gates_in_basis"]
+
+    _flow_unroll = [ConditionalController(_unroll, condition=_unroll_condition)]
```

Comment:
For readability, maybe `_unroll_check` and `_flow_unroll` could be combined like:
```python
_unroll_if_out_of_basis = [
    GatesInBasis(basis_gates),
    ConditionalController(_unroll, condition=_unroll_condition)
]

Reply:
Yes. I think this PR followed the pattern of putting a check in a separate list. But, in this case, it is not necessary.